### PR TITLE
ODS Documentation - Landing Page

### PR DESCRIPTION
**Pull request by JohanSingainy-Moutiapin**
# ODS Documentation - Landing Page

## :bulb: Description

This website is the entry point for the all documentations. There is 2 pages, the first regroups the principal documentations and the second regroups the APIs documentations.
This is finality of documentation harmonisation with OpenDataSoft's identity.

## :green_book: Template

image

image 1

:warning: Template changements : 

* The languages is no more in header but in footer. 
* The items in header are now underlined.
* For the hover animation on the boxes, there is just a background-color changement with the box-shadow growed.

## :wrench: Technical caracteristique

* VueJS : https://vuejs.org/v2/guide/
* Vue-Router : https://router.vuejs.org/en/
* VueX : https://vuex.vuejs.org/en/
* Webpack : https://webpack.js.org/concepts/

## :rocket: Launch application

```shell
npm install

npm run build

npm run dev
```

### CODE REVIEW

**Comment by jmorel** [comment]
*  [ ] Missing translations result in missing content, maybe we could default to english instead?
screen shot 2018-04-06 at 15 58 44
* [ ] On mobile the footer takes up space with no use.
screen shot 2018-04-06 at 15 59 34
* [ ] Invisible end of scrollbar + invisible horizontal scrollbar
screen shot 2018-04-06 at 16 00 36
* [ ] Wrong TPF description
screen shot 2018-04-06 at 16 01 23
* [ ] Missing 404 page
screen shot 2018-04-06 at 16 04 36

**Comment by bost-h** [comment]
How can we test translations ?

File: src/vue/elements/boxes.vue [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+<script>
+export default {
+    name: 'boxes',
+    props: ['lang', 'content', 'img', 'page'],
+    computed: {
+        sizeBox: function () {
+            if (this.page === 'home') return 'ods-container__box-index';
+            else return 'ods-container__box-apis';
+        },
+        sizeIcon: function () {
```
Comment (bost-h) [inline src/vue/elements/boxes.vue]: I think there should be a better solution, at least we could specify the CSS class from the parent component. We can talk about this when you have a moment

File: src/vue/elements/boxes_link.vue [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    </div>
+    
+</template>
+
+<script>
+export default {
+    name: 'boxes_link',
+    props: ['lang', 'content'],
+    computed: {
+        URL : function () {
```
Comment (bost-h) [inline src/vue/elements/boxes_link.vue]: Why is this needed ?

File: src/vue/home/home.vue [hunk 3]
```diff
@@ -0,0 +1,166 @@
+<template>
```
Comment (bost-h) [inline src/vue/home/home.vue]: Looks like there is an unneeded directory here

File: src/vue/page-not-found/page-not-found.vue [hunk 4]
```diff
@@ -0,0 +1,80 @@
+<template>
```
Comment (bost-h) [inline src/vue/page-not-found/page-not-found.vue]: Same, unneeded directory

File: webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+                enforce: 'pre',
+                test: /\.vue$/,
+                loader: 'eslint-loader',
+                exclude: /node_modules/
+            },
+            {
+                test: /\.vue$/,
+                loader: 'vue-loader',
+                options: {
+                    extractCSS: true
```
Comment (bost-h) [inline webpack.config.js]: I know we did it together but maybe CSS extraction should depends on build mode (dev or prod)

File: webpack.config.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+                loader: 'file-loader',
+                options: {
+                    name: '[name].[hash].[ext]'
+                }
+            }
+        ]
+    },
+    plugins: [
+        new HtmlWebpackPlugin({
+            template: './src/index.ejs', // Load a custom template (ejs by default see the FAQ for details),
```
Comment (bost-h) [inline webpack.config.js]: Which FAQ ? HtmlWebpackPlugin README shows examples with HTML files

File: webpack.config.js [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+            minify: minify
+        }),
+        new ExtractTextWebpackPlugin('[name].[contenthash].css')
+    ],
+    resolve: {
+        alias: {
+            'vue$': 'vue/dist/vue.esm.js'
+            // vuex: 'vue/dist/index.esm.js'
```
Comment (bost-h) [inline webpack.config.js]: Comment

File: webpack.config.js [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+        noInfo: true
+    },
+    performance: {
+        hints: false
+    },
+    devtool: '#eval-source-map'
+};
+
+/*
+if (process.env.NODE_ENV === 'development') {
```
Comment (bost-h) [inline webpack.config.js]: It would be nice if we could run the analyzer like `yarn run analyze` (which launch a webpack compilation with the BundleAnalyzerPlugin included)

File: src/js/util.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    /* Open Graph */
+    document.head.querySelector('[property="og:url"]').setAttribute('content', getCurrentURL());
+    document.head.querySelector('[property="og:title"]').setAttribute('content', metas.title);
+    document.head.querySelector('[property="og:description"]').setAttribute('content', metas.description);
+    document.head.querySelector('[property="og:image"]').setAttribute('content', metas.image);
+
+}
+
+export function navigatorLang() {
+    let language = navigator.languages && navigator.languages[0] || // Chrome / Firefox
```
Comment (bost-h) [inline src/js/util.js]: Could you add parenthesis ? It would makes this easier to read, and less bug prone

File: src/js/lang.js [hunk 10]
```diff
@@ -0,0 +1,11 @@
+var store =  {
```
Comment (bost-h) [inline src/js/lang.js]: This seems overengineered :) Personally I would have added `lang` to the root data object and that's all. You can keep the function if you like (but rename it, it is not `setState` is is `setLang`)

File: src/js/lang.js [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+    debug: true,
+    state: {
+        lang: ''
+    },
+    setState (value) {
+        this.state.lang = value;
+    }
+};
+
+module.exports = store;
```
Comment (bost-h) [inline src/js/lang.js]: You should use the new export

File: src/components/elements/boxes.vue [hunk 12]
```diff
@@ -0,0 +1,226 @@
...
+            background-color: white;
+            border: 1px solid lighten(@grey-border, 40%);
+            -webkit-box-shadow: 0px 10px 26px 5px lighten(@grey-border, 15%);
+            -moz-box-shadow: 0px 10px 26px 5px lighten(@grey-border, 15%);
+            box-shadow: 0px 10px 26px 5px lighten(@grey-border, 15%);
+        }
+    }
+}
+
+.ods-container__box-index {
```
Comment (nicoconstanzo) [inline src/components/elements/boxes.vue]: Maybe a `min-height` could be a better fit here? Since the boxes contain text that will be displayed in multiple languages (and therefore different lengths of text which could overflow the box), you should try to avoid such fixed values and use something more flexible.

**Comment by nicoconstanzo** [comment]
On desktop the layout works great, and I really like the transitions you've added to the boxes (where they "float" on hover). 
On smaller screen sizes I've found a coupld issues:
- The `<a class="ods-container__link">` that wraps the blocks is much bigger than the box when in responsive mode, which makes the clickable area much bigger than the content. 
screen shot 2018-04-16 at 18 12 38
- I've noticed that in smaller screen sizes the hover effect is not there although I guess this is voluntary.

**Comment by JohanSingainy-Moutiapin** [comment]
👍 Done for remarks of @bost-h and @nicoconstanzo. Thanks

File: src/components/elements/boxes.vue [hunk 13]
```diff
@@ -0,0 +1,221 @@
+<template>
+
+    <component class="ods-container__link"
+        :class="sizeBox"
+        :is="componentId"
+        v-bind="attributId">
```
Comment (bost-h) [inline src/components/elements/boxes.vue]: I think you could rename this variable since it is not an attribute ID but a link href ?

File: src/components/pages/not-found.vue [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+            setMetas({
+                title:         this.trad.metas.title[this.lang],
+                description:   this.trad.metas.description[this.lang],
+                lang:          this.lang
+            });
+        }
+    },
+    beforeRouteEnter (to, from, next) {
+        next((vm) => {
+            vm.setPageNotFoundMetas();
```
Comment (bost-h) [inline src/components/pages/not-found.vue]: It sounds a bit overkill to declare `setPageNotFoundMetas` as a component public method since it is just used in this place and is not intended to be called from outside

File: src/utils.js [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+
+    /* Open Graph */
+    document.head.querySelector('[property="og:url"]').setAttribute('content', getCurrentURL());
+    document.head.querySelector('[property="og:title"]').setAttribute('content', metas.title);
+    document.head.querySelector('[property="og:description"]').setAttribute('content', metas.description);
+    document.head.querySelector('[property="og:image"]').setAttribute('content', metas.image);
+
+}
+
+// export function navigatorLang() {
```
Comment (bost-h) [inline src/utils.js]: Comments

File: package.json [hunk 16]
```diff
@@ -0,0 +1,52 @@
+{
+  "name": "helphub",
+  "description": "HelpHub Landing Page",
+  "version": "1.0.0",
+  "author": "Johan Singainy-Moutiapin <johan.singainy-moutiapin@opendatasoft.com>",
+  "private": false,
```
Comment (bost-h) [inline package.json]: It should be set as `private: true` to avoid some warning like `No license field` from NPM/Yarn

**Comment by bost-h** [comment]
The translations directory is called `traductions`, which is french

File: .travis.yml [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+    - secure: "hY+6OzPKmFkn47q3ks5d+gJwX4mT8gWTetKls4cNMfvXILu3gdEuNp/oXxOVlgXkdO664LK788nPe7+s7Y/gIgdvxoiG/RH8wrP4w5ha1ciqXlpT4b1U3NHRfjicnFKGLAocBf1GenBgUnW+684LCCUg+XVP9Azm2L+55VcNAiS/clHKZu4L/5akGIFcZO0X53JRwqQXGbvRrbRVPiDQbasZfo8D43fdWj5GwSdrQZedHTfZEES1t5BZtcZV25F9i4u8LCJxmZhg+7qqh6riELw4SsMobLlGvdBETe/NoIjL3oCXPYJbbwFXGo6CccbmworkSNYa4mE8nLNIDvnKy1INxAh3XoTzOHjlCLw45GzQNCaBEcqkjk3QQJkq49ny2Z40Kvs6BjjdDDr9MHBzyaCHyN2Zl53Ok8YhlZNcvc9NCose+7AGGD2F0iiPtCxmDIDTE+FZHNuM4IH/k9eAhEDMbqdDDasUrbpjnDAXBNukyNPYISlPvjmeJeKTLuPUWsHFwQSY4lFZUeLE1SxX+pWFNsDSXEV97ZwysSXSuQU3zarWajXN1o0iPQ95g6N8P7ZAu1deTG17cw+9oyPBXj3qOLgO8UJPLvbUw5lLmCaiNP7hVtQYMm6yKTZSnV6eJcsNTLvkbkQ4FOCjgeH/NWlP4rGibczKy9xZivCnWvg="
```
[the remark this answers is not in the document]
Comment (JohanSingainy-Moutiapin) [reply .travis.yml]: @bost-h This not a duplicate key, but well two different key, the attribut "secure" means that is a secure variable. The variable's name is in the hash.

File: src/components/elements/boxes.vue (hunk 13 above)
Comment (JohanSingainy-Moutiapin) [reply src/components/elements/boxes.vue]: @bost-h Updated !

File: package.json (hunk 16 above)
Comment (JohanSingainy-Moutiapin) [reply package.json]: @bost-h Updated !

**Comment by JohanSingainy-Moutiapin** [comment]
@bost-h, I finished the updates. 👍

File: src/translations/home.js [hunk 18]
```diff
@@ -94,11 +94,11 @@ export default {
             nl: "FAQ/Lexicon"
         },
         desc: {
-            en: "Here you will find answers to your questions about file formats, supported browsers or metadata or simply a word meaning ",
-            fr: "Vous trouverez ici des réponses à vos questions sur les formats de fichiers, les navigateurs pris en charge ou les métadonnées ou simplement une signification de mot",
-            es: "Aquí encontrará las respuestas a sus preguntas sobre formatos de archivos, navegadores admitidos o metadatos, o también qué significa una palabra.",
-            de: "Hier finden Sie Antworten auf Ihre Fragen zu Dateiformaten, unterstützten Browsern oder Metadaten oder die Erklärung einzelner Begriffe",
-            nl: "Hier vindt u antwoorden op uw vragen over bestandsformaten, ondersteunde browsers en metadata of gewoon de betekenis van een woord"
+            en: "Here you will find answers to your questions about file formats, supported browsers, metadata or simply a word meaning.",
+            fr: "Vous trouverez ici des réponses à vos questions sur les formats de fichiers, les navigateurs pris en charge, les métadonnées ou simplement une signification de mot.",
```
Comment (trouba) [inline src/translations/home.js]: "la signification d'un mot." serait plus correct

File: src/translations/home.js [hunk 19]
```diff
@@ -146,11 +146,11 @@ export default {
             nl: "APIs"
         },
         desc: {
-            en: "Our API references, with endpoints, parameters you can use and consoles to run interactive queries",
-            fr: "Nos références d'API, avec les points terminaux, les paramètres que vous pouvez utiliser et les consoles pour exécuter des requêtes interactives",
-            es: "Nuestra referencia de API, con extremos, parámetros que se pueden utilizar y consolas para ejecutar consultas interactivas",
-            de: "Unsere API-Referenzen mit Endpunkten, Parametern, die Sie verwenden können, und Konsolen zum Ausführen interaktiver Abfragen",
-            nl: "Onze API-referenties met eindpunten, parameters die u kunt gebruiken en consoles om interactieve onderzoeken te voeren"
+            en: "Our API references, with endpoints, parameters you can use and consoles to run interactive queries.",
+            fr: "Nos références d'API, avec les points terminaux, les paramètres que vous pouvez utiliser et les consoles pour exécuter des requêtes interactives.",
```
Comment (trouba) [inline src/translations/home.js]: "les consoles pour exécuter des requêtes interactives." 
That doesn't seem to be true
